lib/embedded_ansible.rb

```diff
@@ -64,19 +64,16 @@ def self.services
     AwesomeSpawn.run!("source /etc/sysconfig/ansible-tower; echo $TOWER_SERVICES").output.split
   end
 
-  def self.playbook_extra_variables
-    json_value = {
+  def self.run_setup_script(exclude_tags)
+    json_extra_vars = {
       :minimum_var_space => 0,
       :http_port         => HTTP_PORT,
       :https_port        => HTTPS_PORT
     }.to_json
-    {:e => json_value}
-  end
-  private_class_method :playbook_extra_variables
 
-  def self.run_setup_script(params)
     with_inventory_file do |inventory_file_path|
-      AwesomeSpawn.run!(SETUP_SCRIPT, :params => params.merge(:i => inventory_file_path))
+      params = [[:e, json_extra_vars], [:i, inventory_file_path]] << "--" << ["--skip-tags=", exclude_tags]
```

Comment:
I think this might read easier split out even though you'd get more lines added 😢 

```ruby
params = []
params << [:e, json_extra_vars]
params << [:i, inventory_file_path]
params << "--"
params << ["--skip-tags=", exclude_tags]
```

Otherwise, it might be easy to miss the `"--"`

Also, this way you don't have to see `[` and mentally search for the closing `]`.